packages/tabs/src/styled/StyledTab.ts

```diff
@@ -5,55 +5,118 @@
+import styled, { DefaultTheme, css, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import stripUnit from 'polished/lib/helpers/stripUnit';
 
 const COMPONENT_ID = 'tabs.tab';
 
 interface IStyledTabProps {
-  isHovered?: boolean;
-  isFocused?: boolean;
-  isActive?: boolean;
   disabled?: boolean;
   isSelected?: boolean;
 }
 
 /**
- * Accepts all `<div>` props
+ * 1. A high specificity is needed to apply the border-color in vertical orientations
+ */
+const colorStyles = ({ theme, isSelected }: IStyledTabProps & ThemeProps<DefaultTheme>) => {
+  const selectedColor = getColor('primaryHue', 600, theme);
+
+  return css`
+    border-color: ${isSelected && 'currentColor !important'}; /* [1] */
+    color: ${isSelected ? selectedColor : 'inherit'};
+
+    &:hover {
+      color: ${selectedColor};
+    }
+
+    &:active {
+      border-color: currentColor;
+      color: ${selectedColor};
+    }
+
+    &[data-garden-focus-visible] {
+      color: ${selectedColor};
+    }
+
+    &[data-garden-focus-visible]::before {
+      box-shadow: inset ${theme.shadows.sm(getColor('primaryHue', 600, theme, 0.35)!)};
+    }
+
+    &[aria-disabled='true'] {
+      border-color: transparent;
+      color: ${props => getColor('neutralHue', 400, props.theme)};
+    }
+  `;
+};
+
+const sizeStyles = ({ theme }: ThemeProps<DefaultTheme>) => {
+  const paddingTop = theme.space.base * 2.5;
+  const paddingHorizontal = theme.space.base * 7;
+  const paddingBottom =
+    paddingTop - stripUnit(theme.borderWidths.md) - stripUnit(theme.borderWidths.sm);
+
+  return css`
+    padding: ${paddingTop}px ${paddingHorizontal}px ${paddingBottom}px;
+  `;
+};
+
+/**
+ * 1. Text truncation (requires `max-width`).
+ * 2. Overflow compensation.
  */
 export const StyledTab = styled.div.attrs<IStyledTabProps>(props => ({
   'data-garden-id': COMPONENT_ID,
   'data-garden-version': PACKAGE_VERSION,
-  className: classNames(TabStyles['c-tab__list__item'], {
-    // Visual states
-    [TabStyles['is-hovered']]: props.isHovered,
-    [TabStyles['is-focused']]: props.isFocused,
-    [TabStyles['is-active']]: props.isActive,
-    [TabStyles['is-disabled']]: props.disabled,
-    [TabStyles['is-selected']]: props.isSelected
-  })
+  'aria-disabled': props.disabled
 }))<IStyledTabProps>`
-  /** This removes focus styling when not needed. Mimics :focus-visible */
-  &&& {
-    ${props =>
-      !props.isFocused &&
-      `
-      :before {
-        box-shadow: none;
-      }
-    `}
+  display: inline-block;
+  position: relative;
+  transition: color 0.25s ease-in-out;
+  border-bottom: ${props => props.theme.borderStyles.solid} transparent;
+  border-width: ${props => props.theme.borderWidths.md};
+  cursor: pointer;
+  overflow: hidden; /* [1] */
+  vertical-align: top; /* [2] */
+  user-select: none;
+  text-align: center;
+  text-decoration: none; /* [3] */
```

Comment:
This is looking better, but there is no reference point for `[3]` above.

Reply:
This was an artifact from the `anchor` styling. Removed this line and double-checked the styling.

Reply:
That's no good because now `<Tab as="a" href="...">` is broken:

<img width="903" alt="Screen Shot 2019-11-13 at 2 54 58 PM" src="https://user-images.githubusercontent.com/143773/68811483-9c0ee580-0625-11ea-8697-dc3733873814.png">

As we discussed (offline) that's a scenario we need to retain view styling support for.